codebase/codebase.go

```diff
@@ -0,0 +1,68 @@
+package codebase
+
+import "errors"
+
+// Codebase defines all parameters those are useful to associate Che Editor's window to a WI
+type Codebase struct {
+	Repository string `json:"repository"`
+	Branch     string `json:"branch"`
+	FileName   string `json:"filename"`
+	LineNumber int    `json:"linenumber"`
+}
```

Comment:
Just checking, did we check with Recommender or Che team on this list of params? Curious because I didn't find a story :) 
Sorry if I missed something.

In any case, this list looks fine to start with.